src/components/AppToolbar.tsx

```diff
@@ -134,6 +137,19 @@ class AppToolbarInternal extends React.Component<AppToolbarInternalProps> {
     }
   }
 
+  onFileSelected = (e: File[]) => {
+    const file = e[0];
+    this.props.onLocalPMTilesSelected(file);
+  }
+
+  onFileRejected = (r: FileRejection[]) => {
+    const errorMessageLine = r.map(e => {
```

Comment:
Is it critical to collect all the error messages?

Reply:
I can just show the first error and apply translation for it, if that is okay with you.

Reply:
Are you planning on translating all the errors?
I would simply present a generic error message to the user about failure to open the provided file.

Reply:
I think I can log the errors from react-dropzone to console, and show a translatable error message to the user.